apps/client/src/lib/recherche/queryContents.ts

```diff
@@ -67,13 +67,27 @@ export const filterDispositifs = (
   dispositifs: SimpleDispositif[],
   secondaryThemes: boolean,
   skip: FilterKey | undefined = undefined,
+  allNeeds?: GetNeedResponse[],
 ): SimpleDispositif[] => {
   const filterKeys = buildFilterKeys(query);
   const rule = getDisplayRule(query.type, filterKeys, query.sort);
 
+  const currentThemesInferedBySelectedNeeds: Id[] | undefined =
+    query.themes.length === 0 && query.needs
+      ? allNeeds?.filter((currentNeed) => query.needs.includes(currentNeed._id)).map((need) => need.theme._id)
```

Comment:
Seems like you would need to make sure the resulting theme ids are unique no?

Reply:
Yes, bien vu, merci, j'ai fixé :)